includes/class-wc-taxjar-customer-sync.php

```diff
@@ -146,44 +146,111 @@ public function add_customer_meta_fields( $user ) {
 		endforeach;
 	}
 
+	/**
+     * Saves tax exemption user meta if necessary
+     *
+	 * @param $user_id - Id of user to update
+	 */
 	public function save_customer_meta_fields( $user_id ) {
 		if ( ! apply_filters( 'taxjar_current_user_can_edit_customer_meta_fields', current_user_can( 'manage_woocommerce' ), $user_id ) ) {
 			return;
 		}
 
-		$save_fields = $this->get_customer_meta_fields();
-		$change = false;
-		foreach ( $save_fields as $fieldset ) {
-			foreach ( $fieldset['fields'] as $key => $field ) {
-				if ( isset( $field['type'] ) && 'multi-select' === $field['type'] ) {
-				    $prev_value = get_user_meta( $user_id, $key, true );
-				    if ( empty( $_POST[ $key ] ) ) {
-				        $exempt_regions = '';
-                    } else {
-				        $exempt_regions = array_map( 'wc_clean',  $_POST[ $key ] );
-				        $exempt_regions = implode( ',', $exempt_regions );
-                    }
-
-				    if ( $exempt_regions != $prev_value ) {
-					    update_user_meta( $user_id, $key, $exempt_regions );
-					    $change = true;
-                    }
-				} elseif ( isset( $_POST[ $key ] ) ) {
-					$prev_value = get_user_meta( $user_id, $key, true );
-					$new_value = wc_clean( $_POST[ $key ] );
-					if ( $prev_value != $new_value ) {
-						update_user_meta( $user_id, $key, $new_value );
-						$change = true;
-                    }
-				}
-			}
+		$exemption_settings_changed = false;
+
+		if ( $this->has_exemption_type_changed( $user_id) ) {
+			update_user_meta( $user_id, 'tax_exemption_type', $this->get_posted_exemption_type() );
+			$exemption_settings_changed = true;
+        }
+
+		if ( $this->have_exempt_regions_changed( $user_id ) ) {
+			update_user_meta( $user_id, 'tax_exempt_regions', $this->get_posted_exempt_regions() );
+			$exemption_settings_changed = true;
 		}
 
-		if ( $change ) {
+		if ( $exemption_settings_changed ) {
 		    do_action( 'taxjar_customer_exemption_settings_updated', $user_id );
         }
 	}
 
+	/**
```

Comment:
The following four methods were added as part of the refactoring of the save_customer_meta_fields method. This methods were added to make the logic easier to follow.